salt/states/file.py

```diff
@@ -452,13 +452,16 @@ def _check_directory(name,
                      mode,
                      clean,
                      require,
-                     exclude_pat):
+                     exclude_pat,
+                     max_depth=None):
     '''
     Check what changes need to be made on a directory
     '''
     changes = {}
     if recurse or clean:
-        walk_l = list(os.walk(name))  # walk path only once and store the result
+        assert(max_depth is None or not clean)
```

Comment:
Instead of an assert here, I think it would be better to just cleanly exit and perhaps log an error. This assert will end up just exhibiting itself as a difficult-to-understand error to the user. Thoughts?